### PR TITLE
Add Title prop in SupportButton

### PR DESCRIPTION
**Pull request by GunnerStraiker1**
## Hey, I just made a Pull Request!

This PR is a rebase of the PR #4847 
Add `title` prop in SupportButton component
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core/src/components/SupportButton/SupportButton.tsx [hunk 1]
```diff
@@ -35,6 +35,9 @@ import { SupportItem, SupportItemLink, useSupportConfig } from '../../hooks';
 import { Link } from '../Link';
 
 type Props = {};
+type SupportButtonProps = {
+  title?: string;
+};
```
Comment (freben) [inline packages/core/src/components/SupportButton/SupportButton.tsx:40]: shouldn't this just be one type?

```ts
type SupportButtonProps = {
  title?: string;
  children?: React.ReactNode;
};
```

and then get rid of the `PropsWithChildren` below which is nice

File: packages/core/src/components/SupportButton/SupportButton.tsx [hunk 2]
```diff
@@ -120,6 +126,7 @@ export const SupportButton = ({ children }: PropsWithChildren<Props>) => {
         onClose={popoverCloseHandler}
       >
         <List className={classes.popoverList}>
+          {title && <ListItem alignItems="flex-start">{title}</ListItem>}
```
Comment (freben) [inline packages/core/src/components/SupportButton/SupportButton.tsx:129]: Does this warrant a different text style? Maybe a `<Typography variant="something something"` from here? https://material-ui.com/components/typography/

**Review by freben — COMMENTED** [review]
Thanks for rebuilding this smaller and on top of fresh master!

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx [hunk 3]
```diff
@@ -194,7 +194,9 @@ const CatalogPageContents = (props: CatalogPageProps) => {
               Add example components
             </Button>
           )}
-          <SupportButton>All your software catalog entities</SupportButton>
+          <SupportButton title="Nobody">
```
Comment (Rugvip) [inline plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:197]: This doesn't seem to informative? 😅 Should just move the existing child into the `title` right?

File: packages/core/src/components/SupportButton/SupportButton.tsx [hunk 4]
```diff
@@ -44,6 +43,9 @@ const useStyles = makeStyles(theme => ({
     minWidth: 260,
     maxWidth: 400,
   },
+  title: {
+    fontWeight: 500,
```
Comment (Rugvip) [inline packages/core/src/components/SupportButton/SupportButton.tsx:47]: Is this absolutely needed? Could we use another typography variant instead? It's a bit unfortunate to have overrides like these as it limits peoples ability to provide custom theming.

**Review by Rugvip — COMMENTED** [review]
Thanks!

Iirc the original PR contained a lot more usage of the new `title` prop, I think that could be good to add while at it. Also fine to take care of that later though.

File: plugins/catalog/src/components/CatalogPage/CatalogPage.tsx (hunk 3 above)
Comment (GunnerStraiker1) [reply plugins/catalog/src/components/CatalogPage/CatalogPage.tsx:197]: @Rugvip I forgot to move it😅

**Comment by freben** [comment]
Hey @GunnerStraiker1 - we have done some package rearrangement so now this should be added in `core-components` rather than `core` which is going away soon. Instead of putting that load on everyone to adapt their PRs to, I took the liberty of squashing and shifting this over to the right place. You are marked as a co-author on the commit so your contribution doesn't get lost. As I did this, I also decided to remove the title styling prop, instead favouring a hard coded one.

I hope that's all OK with you. We'll merge later today if this goes green.

**Comment by GunnerStraiker1** [comment]
Yeah sure, no problem with that, I saw also the changes made in the `core` package, so it's ok

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**